### PR TITLE
Bug Fixes / Update states improvement / Update for not connected outputs

### PR DESCRIPTION
**Pull request by granturism0**
Hello Johann
Thanks for your work so far!
Since I use the Dingz - Switch myself, I found a problem in my configuration and made some improvements.

1) Update most of dependencies
2) Update of output states immediately after button pressed
3) Show only output which are connected
4) Bugfix for dimmer 1 status (index 0)

Greetings,
Daniel

### CODE REVIEW

**Comment by johannrichard** [comment]
Hi @granturism0,

Thanks for the PR and the contribution. I'll have a look into it, but it might take some time before I can go through and integrate/close it. Thanks for your patience with that. 

One thing that caught my eye is the `not_connected` value of the `/api/v1/dimmer_config` – I'm quite sure this was not part of the original API but introduced with the `v1.2.x` FIrmware (or in-between the time when I started work on the plugin – as were other changes to the API). I still need to figure out what that means (and if/how I can compare the latest API to an older version 😁). 

In the meantime: would you mind share to post some examples of your config (part. dimmer / input / blinds) so I can understand your particular setup? Thanks.

File: src/dingzAccessory.ts [hunk 1]
```diff
@@ -707,7 +755,7 @@ export class DingzDaAccessory extends EventEmitter {
     newService: Service,
     id: string,
   ) {
-    if (index) {
+    if (index !== null) {
```
Comment (johannrichard) [inline src/dingzAccessory.ts:758]: Nice catch. 👍 😊 😓

File: src/dingzAccessory.ts [hunk 2]
```diff
@@ -561,6 +587,10 @@ export class DingzDaAccessory extends EventEmitter {
             this.platform.log.info(
               `Button ${button} of ${this.device.name} (${service?.displayName}) pressed -> ${action}`,
             );
+
+            // immediately update states after button pressed
+            this.updateDeviceState();
```
Comment (johannrichard) [inline src/dingzAccessory.ts:592]: 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 ---
+## [Released]
+# [1.9.0]
+* Update all dependencies
+* immediately update states after button pressed
+* Fix for updateDimmerState index 0
+
```
Comment (johannrichard) [inline CHANGELOG.md:11]: I usually (but not yet consistently 😅 ) generate the changelog from the commits (which I try to make atomic, i.e. 1 bigger change = 1 commit). I might remove that change from the PR, hope you don't mind.

Comment (granturism0) [reply CHANGELOG.md:11]: No problem and was just a first litter of the Changelog :-)

**Comment by granturism0** [comment]
> Hi @granturism0,
> 
> Thanks for the PR and the contribution. I'll have a look into it, but it might take some time before I can go through and integrate/close it. Thanks for your patience with that.
> 
> One thing that caught my eye is the `not_connected` value of the `/api/v1/dimmer_config` – I'm quite sure this was not part of the original API but introduced with the `v1.2.x` FIrmware (or in-between the time when I started work on the plugin – as were other changes to the API). I still need to figure out what that means (and if/how I can compare the latest API to an older version 😁).
> 
> In the meantime: would you mind share to post some examples of your config (part. dimmer / input / blinds) so I can understand your particular setup? Thanks.

Hallo @johannrichard
no problem and take your time for the review. I have been running this code for a few days now and for me it fits perfectly so far and the bugs are gone now.
If the "not_connected" was in it before version 1.2.x is not known to me and I don't find this old API anymore. If you want to be on the safe side, we would have to check the API version in the code and check for >1.2.

On my test switch I use the following config:
Switch 1 controls output 1 where I have a non dimmable light on. Output two is deactivated, but I use switch two to control a http-address.
Switch 3 and 4 are occupied by the blind motor.
If you want to test other configs, let me know.

File: src/dingzAccessory.ts (hunk 1 above)
Comment (granturism0) [reply src/dingzAccessory.ts:758]: Yes, now I can see output 1, which is index = 0 ;-)

File: src/dingzAccessory.ts [hunk 4]
```diff
@@ -409,7 +395,12 @@ export class DingzDaAccessory extends EventEmitter {
     switch (this.dingzDeviceInfo.dip_config) {
       case 3:
         // DIP = 0: D0, D1, D2, D3; (Subtypes) (Unless Input, then D1, D2, D3)
-        if (inputConfig && !inputConfig[0].active) {
+        if (
+          inputConfig &&
+          !inputConfig[0].active &&
+          dimmerConfig?.dimmers[0].output &&
+          dimmerConfig?.dimmers[0].output !== 'not_connected'
```
Comment (johannrichard) [inline src/dingzAccessory.ts:402]: Note to self: `/api/v1/state` is our friend here: `not_connected` is similar to the `input` being active: both make that the dimmer is set as `readonly` whereas the DIP switch applies on a more fundamental level. 😅

**Review by johannrichard — APPROVED** [review]
All fine.

**Comment by johannrichard** [comment]
> If the "not_connected" was in it before version 1.2.x is not known to me and I don't find this old API anymore. If you want to be on the safe side, we would have to check the API version in the code and check for >1.2.

It most certainly wasn't otherwise I would have added it to the type definition already. However, its not a *breaking* change inso far as you can only set it with newer FIrmwares.

**Pull request merged**